### PR TITLE
Add simplified, independent version of ad renderer from internal code base

### PR DESCRIPTION
**Pull request by tlauinger**

### CODE REVIEW

File: fb_ad_creative_retriever.py [hunk 1]
```diff
@@ -501,7 +494,11 @@ def process_fetched_ad_creative_data(self, archive_id, fetched_data):
             video_bucket_path = None
             if creative.image:
                 try:
-                    image_dhash = get_image_dhash(creative.image.binary_data)
+                    if creative.image.binary_data in (None, "", b""):
+                        # this can happen when missing ad images are allowed (e.g., for testing with old ads)
+                        image_dhash = "00000000000000000000000000000000"
```
Comment (macpd) [inline fb_ad_creative_retriever.py:499]: I'd like to keep this `None` so it's `null` in the database. makes it easier to determine if there was an image found

**Review by macpd — APPROVED** [review]
Thank you so much for doing this. just one nit about keeping `image_dhash` if no image found. otherwise everything else looks great.

File: fb_ad_creative_retriever.py (hunk 1 above)
Comment (tlauinger) [reply fb_ad_creative_retriever.py:499]: Reading the code in more detail, it seems the exception clause already handles the case of empty images, too, so in theory the emptiness check could be removed. But then the `continue` will cause ad creatives with invalid/missing images to be skipped entirely, and I'm not sure that's intended behaviour. At least for the test cases where we know the image isn't available any more, skipping the creative entirely will cause the test to fail. How would you like to handle corrupted/missing images for non-test scenarios?

One option could be to "break" out of the if statement only (i.e., only skip uploading of corrupted/missing images but still process the text).

Comment (macpd) [reply fb_ad_creative_retriever.py:499]: Ah, that makes sense. I forgot about the missing/corrupted image case. With that in mind I think the current behavior is fine.

**Review by macpd — APPROVED** [review]
Please squash and merge whenever you're ready. Thanks again for doing this.

**Pull request merged**